templates/core/terraform/appgateway/appgateway.tf

```diff
@@ -18,83 +24,166 @@ resource "azurerm_application_gateway" "agw" {
+    ssl_certificate_name           = local.certificate_name
+  }
+
+  # Public HTTP listener
+  http_listener {
+    name                           = local.insecure_listener_name
+    frontend_ip_configuration_name = local.frontend_ip_configuration_name
+    frontend_port_name             = local.insecure_frontend_port_name
+    protocol                       = "Http"
   }
 
   request_routing_rule {
     name                       = local.request_routing_rule_name
     rule_type                  = "PathBasedRouting"
-    http_listener_name         = local.listener_name
-    url_path_map_name          = local.management_api_url_path_map_name_pool_name
+    http_listener_name         = local.secure_listener_name
+    url_path_map_name          = local.app_path_map_name
+  }
+
+  # Routing rule to redirect non-secure traffic to HTTPS
+  request_routing_rule {
+    name                       = local.redirect_request_routing_rule_name
+    rule_type                  = "PathBasedRouting"
+    http_listener_name         = local.insecure_listener_name
+    url_path_map_name          = local.redirect_path_map_name
   }
 
+  # Default traffic is routed to the static website. Exception is API.
   url_path_map {
-    name = local.management_api_url_path_map_name_pool_name
-    default_backend_address_pool_name  = local.backend_address_pool_name
-    default_backend_http_settings_name = local.http_setting_name
+    name = local.app_path_map_name
+    default_backend_address_pool_name  = local.staticweb_backend_pool_name
+    default_backend_http_settings_name = local.staticweb_http_setting_name
 
     path_rule {
       name = "api"
-      paths = ["/api/*"]
-      backend_address_pool_name = local.management_api_backend_address_pool_name
-      backend_http_settings_name = local.http_setting_name
+      paths = [ "/api/*", "/docs", "/openapi.json" ]
+      backend_address_pool_name = local.api_backend_pool_name
+      backend_http_settings_name = local.api_http_setting_name
+    }
+  }
+
+  # Redirect any HTTP traffic to HTTPS unless its the ACME challenge path used for LetsEncrypt validation.
+  url_path_map {
+    name = local.redirect_path_map_name
+    default_redirect_configuration_name = local.redirect_configuration_name
+
+    path_rule {
+      name = "acme"
+      paths = ["/.well-known/acme-challenge/*"]
+      backend_address_pool_name = local.staticweb_backend_pool_name
+      backend_http_settings_name = local.staticweb_http_setting_name
     } 
   }
+
+  # Redirect to HTTPS
+  redirect_configuration {
+    name = local.redirect_configuration_name
+    redirect_type = "Permanent"
+    target_listener_name = local.secure_listener_name
+    include_path = true
+    include_query_string = true
+  }
+
+  # We don't want Terraform to revert certificate cycle changes. We assume the certificate will be renewed in keyvault.
+  lifecycle {
+    ignore_changes = [
+      ssl_certificate
+    ]
+  }
+
 }
 
 data "azurerm_public_ip" "appgwpip_data" {
```

Comment:
`make all` fails to setup this one:

	Error: Public IP "pip-agw-limorltre-lldev-9808" (Resource Group "rg-limorltre-lldev-9808") was not found
	
It also expects the certificate in keyvault, yet according to the (offline) guidance, male letsencrypt should be run afterwards. It's a confusing dev experience IMO.

Reply:
@limorl , please see #94 which captures above error

Reply:
I'll mark this as resolved here, as this PR doesn't change the IP definition. We should try to resolve this error as as fix for #94.